common/functions/src/scalars/others/inet_ntoa.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Datafuse Lfloor.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::str;
+
+use common_datavalues::prelude::DataColumn;
+use common_datavalues::prelude::DataColumnsWithField;
+use common_datavalues::DataSchema;
+use common_datavalues::DataType;
+use common_datavalues::DataValue;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+
+#[derive(Clone)]
+pub struct InetNtoaFunction {
+    display_name: String,
+}
+
+impl InetNtoaFunction {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(InetNtoaFunction {
+            display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic())
+    }
+}
+
+impl Function for InetNtoaFunction {
+    fn name(&self) -> &str {
+        &*self.display_name
+    }
+
+    fn num_arguments(&self) -> usize {
+        1
+    }
+
+    fn return_type(&self, args: &[DataType]) -> Result<DataType> {
```

Comment:
Let's accept more datatypes as arg, see:  https://www.db-fiddle.com/f/hFEuh2zxZY35eSxsmaaRDv/0

  

```
if is_numeric(&args[0]) || args[0] == DataType::String {
            Ok(DataType::String)
        } else {
            Err(ErrorCode::IllegalDataType(format!(
                "Expected numeric, but got {}",
                args[0]
            )))
        }
```

Reply:
Thanks for suggestions. inet_ntoa support null, string, numerics types

Reply:
I add these test cases

```
        Test {
            name: "inet_ntoa('323a111')",
            func: InetNtoaFunction::try_create("inet_ntoa('323a111')")?,
            arg: DataColumnWithField::new(
                Series::new(["323a111"]).into(),
                DataField::new("arg1", DataType::UInt32, false),
            ),
            expect: Ok(DataColumn::Constant(
                DataValue::String(Some(b"0.0.1.67".to_vec())),
                1,
            )),
        },
        Test {
            name: "inet_ntoa('-1')",
            func: InetNtoaFunction::try_create("inet_ntoa('323a111')")?,
            arg: DataColumnWithField::new(
                Series::new(["-1"]).into(),
                DataField::new("arg1", DataType::UInt32, false),
            ),
            expect: Ok(DataColumn::Constant(
                DataValue::Null,
                1,
            )),
        },
```